src/Cli/dotnet/commands/dotnet-workload/search/versions/SearchWorkloadSetsCommand.cs

```diff
@@ -0,0 +1,87 @@
+        _workloadSetOutputFormat = result.GetValue(SearchWorkloadSetsParser.FormatOption);
+        workloadResolverFactory = workloadResolverFactory ?? new WorkloadResolverFactory();
+        var creationResult = workloadResolverFactory.Create();
+
+        SdkVersion = creationResult.SdkVersion;
+        _workloadResolver = creationResult.WorkloadResolver;
+
+        Installer = WorkloadInstallerFactory.GetWorkloadInstaller(
+                reporter,
+                new SdkFeatureBand(SdkVersion),
+                _workloadResolver,
+                Verbosity,
+                creationResult.UserProfileDir,
+                !SignCheck.IsDotNetSigned(),
+                restoreActionConfig: new RestoreActionConfig(result.HasOption(SharedOptions.InteractiveOption)),
+                elevationRequired: false,
+                shouldLog: false);
+    }
+
+    protected override void ShowHelpOrErrorIfAppropriate(ParseResult parseResult)
+    {
+
+    }
+
+    public async Task<int> Execute(CancellationToken cancellationToken)
+    {
+        var featureBand = new SdkFeatureBand(SdkVersion);
+        var packageId = Installer.GetManifestPackageId(WorkloadPackageIdBase, featureBand);
+        var versions = (await PackageDownloader.GetLatestPackageVersions(packageId, NumberOfWorkloadSetsToTake, packageSourceLocation: null, includePreview: !string.IsNullOrWhiteSpace(SdkVersion.Prerelease)).ConfigureAwait(false))
+            .Select(version => WorkloadManifestUpdater.WorkloadSetPackageVersionToWorkloadSetVersion(featureBand, version.Version.ToString()));
+        if (_workloadSetOutputFormat == SearchWorkloadSetsFormat.json)
+        {
+            Reporter.WriteLine(JsonSerializer.Serialize(versions.Select(version => version.ToDictionary(_ => "workloadVersion", v => v))));
+        }
+        else
+        {
+            Reporter.WriteLine(string.Join('\n', versions));
+        }
+
+        return 0;
+    }
+
+    public override int Execute() => Execute(CancellationToken.None).GetAwaiter().GetResult();
+
+    /// <summary>
+    /// Used for CLI completions to get the workload set versions.
+    /// </summary>
+    /// <returns></returns>
+    public static async Task<IEnumerable<string>> GetWorkloadSetVersions(CompletionContext ctx)
+    {
+        var inner = new SearchWorkloadSetsCommand(ctx.ParseResult);
+        var featureBand = new SdkFeatureBand(inner.SdkVersion);
+        var packageId = inner.Installer.GetManifestPackageId(WorkloadPackageIdBase, featureBand);
+        var versions = (await inner.PackageDownloader.GetLatestPackageVersions(packageId, inner.NumberOfWorkloadSetsToTake, packageSourceLocation: null, includePreview: !string.IsNullOrWhiteSpace(inner.SdkVersion.Prerelease)).ConfigureAwait(false))
+            .Select(version => WorkloadManifestUpdater.WorkloadSetPackageVersionToWorkloadSetVersion(featureBand, version.Version.ToString()));
+        return versions;
+    }
```

Comment:
This is a simple static accessor to the same core logic in `Execute`, for tab completion purposes.

Reply:
I do wonder if we can make a helper method to reuse more of this; though most of the names of things are different, the logic is mostly the same